app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -258,20 +274,41 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
                 }
     }
 
+    private fun maybeOpenQueuedTab(): Disposable? {
+        val queuedTab = serviceLocator!!.fxaRepo.queuedTab ?: return null
+
+        // This is a really bad solution, and could cause future race conditions. Executing
+        // immediately throws "java.lang.IllegalStateException: FragmentManager is already
+        // executing transactions", and I couldn't find any better way to deconflict
+        return Observable.just("unused")
+            .delay(1, TimeUnit.SECONDS)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { receiveTab(queuedTab) }
```

Comment:
I really hate this.  This is the worst part of the PR.  I couldn't work out any other solution, but if you have any suggestions I am happy to rewrite it.

Reply:
Good call, that fixed it.  Thanks!